Support setting `reportDir` for `MochaReporter`

cqcmdwym opened the pull request:
Related with Issue : https://github.com/microsoft/FluidFramework/issues/6714

cqcmdwym committed:
Support setting `reportDir` for `MochaReporter`

DLehenbauer commented on tools/benchmark/src/MochaReporter.ts:
```diff
@@ -42,8 +42,9 @@ function getName(name: string): string {
-    public constructor(runner: Runner) {
-        const benchmarkReporter = new BenchmarkReporter();
+    public constructor(runner: Runner, options?: ReporterOptions) {
+        reporters.Base.call(this, runner, options);
```
This looks straight-forward except for invoking the cctor function for `reporters.Base`.

@cqcmdwym - I was wondering if it would work to have this class extend 'reporters.Base' and call `super(runner, options)`?

cqcmdwym replied:
yes, if call super(runner, options),it will inherit mocha report power, however, maybe here needn't.

DLehenbauer commented on tools/benchmark/src/Reporter.ts:
```diff
@@ -113,7 +113,7 @@ export class BenchmarkReporter {
         // If changing this or the result file logic in general,
         // be sure to update the glob used to look for output files in the perf pipeline.
-        this.outputDirectory = outputDirectory ?? path.join(__dirname, ".output");
+        this.outputDirectory = outputDirectory ? path.join('.', outputDirectory) : path.join(__dirname, ".output");
```
@cqcmdwym - I was wondering why this prepends './' to the output directory?  I think this prevents users from using absolute paths (e.g., '/tmp/foo').

cqcmdwym replied:
@DLehenbauer Yes, I also have this concern when I thinking. I prefer to pass a relative path when using this option.
Do you have any other opinion? Thanks.

cqcmdwym replied:
I have use path.resolve function to handle both of the absolute and relative path.

DLehenbauer reviewed (changes requested):
@cqcmdwym - Thanks for adding the 'reportDir' option!  I had a couple quick questions...  (see PR comments.  Thx!)

@CraigMacomber - Did you have any feedback?

cqcmdwym committed:
Remove unnecessary call per comment

cqcmdwym committed:
Fix linter issue

cqcmdwym committed:
Fix lint

cqcmdwym committed:
resolve path

DLehenbauer commented on tools/benchmark/src/Reporter.ts:
```diff
@@ -113,7 +113,8 @@ export class BenchmarkReporter {
         // be sure to update the glob used to look for output files in the perf pipeline.
-        this.outputDirectory = outputDirectory ?? path.join(__dirname, ".output");
+        // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+        this.outputDirectory = outputDirectory ? path.resolve(process.cwd(), outputDirectory) : path.join(__dirname, ".output");
```
This is a good solution, thanks!

DLehenbauer committed:
Create Reporter.ts

I like the approach of using process.resolve()!

I don't think you need to pass 'process.cwd()', though, as process.resolve() will implicitly resolve against the current working directory if the path is relative.

DLehenbauer reviewed (approved):
Looks good.  Thanks, @cqcmdwym !
